gen/aircraft.py

```diff
@@ -787,8 +795,12 @@ def _setup_group(
             OptReactOnThreat(OptReactOnThreat.Values.EvadeFire)
         )
 
-        channel = self.get_intra_flight_channel(unit_type)
+        if flight.flight_type == FlightType.AEWC:
+            channel = self.radio_registry.alloc_uhf()
+        else:
+            channel = self.get_intra_flight_channel(unit_type)
         group.set_frequency(channel.mhz)
+        self.channel = channel
```

Comment:
This should not be a member. Just create the `AwacsInfo` here so you don't need to pass the data back.